File: RASPY_Files_and_Backend/backend/app/models/announcement.py
from sqlalchemy import Column, Integer, String, Boolean, ForeignKey, Text, DateTime
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from app.database import Base

class Announcement(Base):
    __tablename__ = "tbl_announcement"

    announcement_id = Column(Integer, primary_key=True, autoincrement=True)
    title = Column(String(255), nullable=False)
    content = Column(Text, nullable=False)
    date_posted = Column(DateTime, server_default=func.now())
    is_published = Column(Boolean, default=True)
    attachment_path = Column(String(500))
    created_by = Column(Integer, ForeignKey("tbl_users.user_id", ondelete="CASCADE"), nullable=False)

    creator = relationship(
        "User",
        primaryjoin="Announcement.created_by == User.user_id"
    )

File: RASPY_Files_and_Backend/frontend/flutter_app/lib/models/announcement.dart
class Announcement {
  final int announcementId;
  final String title;
  final String content;
  final String? datePosted;
  final bool isPublished;
  final String? attachmentPath;

  Announcement({
    required this.announcementId,
    required this.title,
    required this.content,
    this.datePosted,
    this.isPublished = true,
    this.attachmentPath,
  });

  factory Announcement.fromJson(Map<String, dynamic> json) {
    return Announcement(
      announcementId: json['announcement_id'] is int
          ? json['announcement_id']
          : int.tryParse(json['announcement_id']?.toString() ?? '') ?? 0,
      title: json['title']?.toString() ?? '',
      content: json['content']?.toString() ?? '',
      datePosted: json['date_posted']?.toString(),
      isPublished: json['is_published'] is bool
          ? json['is_published']
          : (json['is_published'] == 1 || json['is_published'] == 'true'),
      attachmentPath: json['attachment_path']?.toString(),
    );
  }
}